File: pokedex/lib/models/pokemon_basic.dart
class PokemonBasic {
  int id;
  String name;
  String picture;
  double attack;
  double defense;

  PokemonBasic(this.id, this.name, this.attack, this.defense, this.picture);

  PokemonBasic.fromJson(Map json) {
    this.id = json['id'];
    this.name = json['name'];
    this.attack = json['attack'];
    this.defense = json['defense'];
    this.picture = json['picture'];
  }

  @override
  bool operator ==(o) => o is PokemonBasic && name == o.name && id == o.id;

  @override
  int get hashCode => name.hashCode ^ id.hashCode;
}

File: pokedex-api/src/main/java/fr/pokedex/model/PokemonBasic.java
package fr.pokedex.model;

public class PokemonBasic extends AbstractPokemon {
  public String picture;

  public PokemonBasic(int id, String name, double attack, double defense, String picture) {
    super(id, name, attack, defense);
    this.picture = picture;
  }

  public double getAttack() {
    return attack;
  }

  public double getDefense() {
    return defense;
  }

  public String getName() {
    return name;
  }

  public int getId() {
    return id;
  }
}
